### ISSUE TITLE
Cannot make nREPL work with LT

### ISSUE THREAD

**Issue by jmatsushita** [bhauman/lein-figwheel#119]
Hi there,

I've been following this om tutorial https://github.com/omcljs/om/wiki/Basic-Tutorial and tried to updated the project.clj with the 0.2.5 versions of figwheel and lein-figwheel and removed the commented line ```:nrepl-port 7888``` but it doesn't seem that the nREPL is starting (can't see anything in the console, in the log, and can't connect to 7888 from lighttable).

Cheers,

Jun

**Comment by bensu** [comment]
I'll take a look. Do you have Light Table's last version?

**Comment by bensu** [comment]
Also, did the connection work before you updated your dependencies?

**Comment by jmatsushita** [comment]
I hadn't properly referenced the lein-light-repl dependencies and nrepl-middleware.

Now I get the following exception (same as in #11) when starting lein repl (from an updated chestnut template) 
```
Error loading lighttable.nrepl.handler: java.lang.RuntimeException: No such var: fs/*cwd*, compiling:(lighttable/nrepl/core.clj:21:30)
Exception in thread "main" java.lang.RuntimeException: Unable to resolve var: lighttable.nrepl.handler/lighttable-ops in this context, compiling:(/private/var/folders/qr/3s9y_9dn3679_sfvk80swjb40000gr/T/form-init7332411806645597825.clj:1:2678)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6651)
	at clojure.lang.Compiler.analyze(Compiler.java:6445)
	at clojure.lang.Compiler.analyze(Compiler.java:6406)
	at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3719)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6646)
	at clojure.lang.Compiler.analyze(Compiler.java:6445)
	at clojure.lang.Compiler.analyze(Compiler.java:6406)
	at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3719)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6646)
	at clojure.lang.Compiler.analyze(Compiler.java:6445)
	at clojure.lang.Compiler.access$100(Compiler.java:38)
	at clojure.lang.Compiler$LetExpr$Parser.parse(Compiler.java:6050)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6644)
	at clojure.lang.Compiler.analyze(Compiler.java:6445)
	at clojure.lang.Compiler.analyze(Compiler.java:6406)
	at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:5782)
	at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5217)
	at clojure.lang.Compiler$FnExpr.parse(Compiler.java:3846)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6642)
	at clojure.lang.Compiler.analyze(Compiler.java:6445)
	at clojure.lang.Compiler.eval(Compiler.java:6700)
	at clojure.lang.Compiler.eval(Compiler.java:6693)
	at clojure.lang.Compiler.eval(Compiler.java:6693)
	at clojure.lang.Compiler.load(Compiler.java:7130)
	at clojure.lang.Compiler.loadFile(Compiler.java:7086)
	at clojure.main$load_script.invoke(main.clj:274)
	at clojure.main$init_opt.invoke(main.clj:279)
	at clojure.main$initialize.invoke(main.clj:307)
	at clojure.main$null_opt.invoke(main.clj:342)
	at clojure.main$main.doInvoke(main.clj:420)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.RuntimeException: Unable to resolve var: lighttable.nrepl.handler/lighttable-ops in this context
	at clojure.lang.Util.runtimeException(Util.java:221)
	at clojure.lang.Compiler$TheVarExpr$Parser.parse(Compiler.java:659)
	at clojure.lang.Compiler.analyzeSeq(Compiler.java:6644)
	... 34 more
Exception in thread "Thread-3" clojure.lang.ExceptionInfo: Subprocess failed {:exit-code 1}
	at clojure.core$ex_info.invoke(core.clj:4403)
	at leiningen.core.eval$fn__6725.invoke(eval.clj:236)
	at clojure.lang.MultiFn.invoke(MultiFn.java:231)
	at leiningen.core.eval$eval_in_project.invoke(eval.clj:337)
	at leiningen.repl$server$fn__10717.invoke(repl.clj:241)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invoke(core.clj:624)
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1862)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invoke(core.clj:628)
	at clojure.core$bound_fn_STAR_$fn__4140.doInvoke(core.clj:1884)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.lang.Thread.run(Thread.java:745)
```

My project.clj is:
```
(defproject my-project "0.1.0-SNAPSHOT"
  :description ""
  :url ""
  :license {:name "GNU General Public License"
            :url "http://www.gnu.org/licenses/gpl-3.0.html"}

  :source-paths ["src/clj"]
  :repl-options {:timeout 200000} ;; Defaults to 30000 (30 seconds)

  :test-paths ["spec/clj"]

  :dependencies [[org.clojure/clojure "1.6.0"]
                 [org.clojure/clojurescript "0.0-3058" :scope "provided"]
                 [ring "1.3.2"]
                 [ring/ring-defaults "0.1.4"]
                 [compojure "1.3.2"]
                 [enlive "1.1.5"]
                 [org.omcljs/om "0.8.8"]
                 [environ "1.0.0"]
                 [http-kit "2.1.19"]
                 [prismatic/om-tools "0.3.11"]
                 [prismatic/dommy "1.0.0"]
                 [sablono "0.3.5-SNAPSHOT"]
                 [ankha "0.1.4"]
                 [secretary "1.2.3"]
                 [lein-light-nrepl "0.1.0"]]

  :plugins [[lein-cljsbuild "1.0.5"]
            [lein-environ "1.0.0"]]

  :min-lein-version "2.5.0"

  :uberjar-name "my-project.jar"

  :cljsbuild {:builds {:app {:source-paths ["src/cljs"]
                             :compiler {:output-to     "resources/public/js/app.js"
                                        :output-dir    "resources/public/js/out"
                                        :source-map    "resources/public/js/out.js.map"
                                        :optimizations :none
                                        :pretty-print  true}}}}

  :profiles {:dev {:source-paths ["env/dev/clj"]
                   :test-paths ["test/clj"]

                   :dependencies [[figwheel "0.2.5"]
                                  [figwheel-sidecar "0.2.5"]
                                  [com.cemerick/piggieback "0.1.5"]
                                  #_[weasel "0.6.0"]
                                  [speclj "3.2.0"]]

                   :repl-options {:init-ns my-project.server
                                  :nrepl-middleware [lighttable.nrepl.handler/lighttable-ops]}

                   :plugins [[lein-figwheel "0.2.5"]
                             [speclj "3.2.0"]]

                   :figwheel {:http-server-root "public"
                              :server-port 3449
                              :css-dirs ["resources/public/css"]}

                   :env {:is-dev true}

                   :cljsbuild {:test-commands { "spec" ["phantomjs" "bin/speclj" "resources/public/js/app_test.js"] }
                               :builds {:app {:source-paths ["src/cljs" "env/dev/cljs"]}
                                        :test {:source-paths ["src/cljs" "spec/cljs"]
                                               :compiler {:output-to     "resources/public/js/app_test.js"
                                                          :output-dir    "resources/public/js/test"
                                                          :source-map    "resources/public/js/test.js.map"
                                                          :optimizations :whitespace
                                                          :pretty-print  false}}}}}

             :uberjar {:source-paths ["env/prod/clj"]
                       :hooks [leiningen.cljsbuild]
                       :env {:production true}
                       :omit-source true
                       :aot :all
                       :cljsbuild {:builds {:app
                                            {:source-paths ["env/prod/cljs"]
                                             :compiler
                                             {:optimizations :advanced
                                              :pretty-print false}}}}}})
```

**Comment by bensu** [comment]
This still looks like a dependency conflict. LightTable is not moving forward dependency wise, i.e. `[clojure "1.5.1"]` & `[clojurescript "0.0-2138"]`. To troubleshoot this further, try `lein deps :tree` to find where if fs has a conflict. LightTable is using `[fs "1.3.3"]` while the last version is `"1.4.6"`

**Comment by jmatsushita** [comment]
Thanks for the hint I think you're right. However actually my ```lein deps :tree``` shows that ```cljsbuild``` depends on ```["fs 1.1.2"]``` so it might not be LightTable...
```
 [figwheel-sidecar "0.2.5" :scope "test"]
   ...
   [cljsbuild "1.0.4" :scope "test"]
     [fs "1.1.2" :scope "test"]
   ...
```

**Comment by jmatsushita** [comment]
Ok, so I solved it by using the [lein checkout dependencies](https://github.com/technomancy/leiningen/blob/master/doc/TUTORIAL.md#checkout-dependencies) method and creating local versions of ```figwheel-sidecar``` and ```cljsbuild```. I'm new to this, so I don't know if there's a more elegant way. In any case, I'll post a request for updating the dependency upstream to cljsbuild. I guess we should keep this open until both cljsbuild and figwheel bump up?

**Comment by bensu** [comment]
I agree, leave this open. I declared LightTable guilty when it was innocent, but it is still going to be a an increasing problem in the future. ClojureScript is moving terribly fast compared to LightTable

**Comment by bhauman** [comment]
Oh yeah this got reintroduced when I fixed #72.  

Darn. I did a lot of work to keep cljsbuild out of the running env only to forget and reintroduce it. In order to get it back out I'd have to steal all of the process launching code to handle `:notify-command`.  Yuck.

**Comment by bhauman** [comment]
I have removed cljsbuild from the dependencies and I think that this was one of the big deal breakers.  I'm going to close this.
